tests/probes/lttng-test.c

```diff
@@ -94,13 +122,270 @@ static const struct file_operations lttng_test_filter_event_operations = {
 	.write = lttng_test_filter_event_write,
 };
 
+/**
+ * event_a: triggers $arg lttng_test_event_a events
+ * @arg: string containing the number of event to generate
+ * @id: Unique identifier of the caller
+ *
+ * Returns 0 on success and a negative value on parsing error
+ */
+static int event_a(char *arg, int id)
+{
+	int i, ret;
+	unsigned long iter;
+
+	ret = kstrtoul(arg, 10, &iter);
+	if (ret) {
+		printk(KERN_ERR "Failed to parse %s arg", __func__);
+		goto error;
+	}
+
+	if (iter > LTTNG_TEST_EVENT_MAX) {
+		printk(KERN_ERR "Number of iteration (%lu) exceeding maximum ", iter);
+		ret = -EDOM;
+		goto error;
+	}
+
+	ret = 0;
+	for (i = 0; i < iter; i++) {
+		trace_lttng_test_event_a(id);
+	}
+
+error:
+	return ret;
+}
+
+/**
+ * event_b: triggers $arg lttng_test_event_b events
+ * @arg: string containing the number of event to generate
+ * @id: Unique identifier of the caller
+ *
+ * Returns 0 on success and a negative value on parsing error
+ */
+static int event_b(char *arg, int id)
+{
+	int i, ret;
+	unsigned long iter;
+
+	ret = kstrtoul(arg, 10, &iter);
+	if (ret) {
+		printk(KERN_ERR "Failed to parse %s arg", __func__);
+		goto error;
+	}
+
+	if (iter > LTTNG_TEST_EVENT_MAX) {
+		printk(KERN_ERR "Number of iteration (%lu) exceeding maximum ", iter);
+		ret = -EDOM;
+		goto error;
+	}
+
+	ret = 0;
+	for (i = 0; i < iter; i++) {
+		trace_lttng_test_event_b(id);
+	}
+
+error:
+	return ret;
+}
+
+/**
+ * Array of method descriptions.
+ */
+static struct lttng_test_method test_methods[] = {
+	{"event_a", "count", &event_a},
+	{"event_b", "count", &event_b},
+};
+
+/*
+ * Returns the number of space-delimited words in a given string. Returns -1, if
+ * non-ascii byte is found.
+ */
+static int count_word_in_str(char *str, size_t len)
+{
+	int i, count, prev_is_space;
+
+	count = 0;
+	i = 0;
+	prev_is_space = 1;
+	while (i < len && str[i] != '\0') {
```

Comment:
I think it would be clearer like that:
for (i = 0; i < len && str[i] != '\0'; i++)